src/StreamReader.js

```diff
@@ -48,7 +48,15 @@ const METADATA_REGEX = /(\w+)=['"](.+?)['"];/g;
  */
 const _parseMetadata = metadata => {
   const data = Buffer.isBuffer(metadata) ? metadata.toString('utf8') : metadata || '';
-  const parts = [...data.replace(/\0*$/, '').matchAll(METADATA_REGEX)];
+  const parts = [];
+  (function getParts () {
```

Comment:
We don't need an IIFE here, just leave the `string` and everything in the function in the same scope where `parts` is.